scripts/update_workspace_dd.py

```diff
@@ -450,6 +457,21 @@ def get_replacement_path(original_path, mapping):
         fail_reason: if no new_path, more information about failure
     '''
 
+    def get_destination_from_mapping(path):
```

Comment:
I generally try to avoid functions-in-functions when I can, though I'm not entirely sure that's a hard and fast rule. But it caught my eye because `mapping` isn't an input to this function... is there a reason not to have `get_destination_from_mapping` defined outside of `get_replacement_path`, and pass `mapping` to it as an input along with `path`?